src/Moq/Mock.cs

```diff
@@ -687,9 +687,10 @@ private static void SetupAllPropertiesPexProtected(Mock mock, DefaultValueProvid
 							initialValue = mock.GetDefaultValue(getter, useAlternateProvider: DefaultValueProvider.Empty);
 						}
 
-						if (initialValue is IMocked mocked)
+						if (Mock.TryGetFromReturnValue(initialValue, out var innerMock))
```

Comment:
Up until now, `SetupAllProperties` didn't recursively set up inner mocks yielded by properties of type `[Value]Task<TResult>`. Granted, such properties may be seen as an API design error, but this is still a functional change that should perhaps be documented in a unit test.